src/parameter/parameterextractor.cpp

```diff
@@ -37,33 +37,27 @@ void ParameterExtractor::setParameters(const FileModule* module, bool rebuildPar
     const ValuePtr defaultValue = assignment.expr->evaluate(&ctx);
     if (defaultValue->type() == Value::ValueType::UNDEFINED) continue;
 
-    ParameterObject *entryObject = new ParameterObject();
-    entryObject->setAssignment(&ctx, &assignment, defaultValue);
+    ParameterObject *entryObject = new ParameterObject(&ctx, &assignment, defaultValue);
```

Comment:
Is there a way we could make this e.g. a `shared_ptr`, to get rid of some error prone memory management?

Reply:
good idea - that would resolve both memory leaks (caused by not deleting) and race conditions (caused by deleting in one UI event, while an other event is waiting to be processed).

added to #2316 as this is not straight forward.